### ISSUE TITLE
Develop an "all-in-one" environment for development and testing

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#10]
It is pretty clear that both for development and for simple example use / kick the tires testing we need an all-in-one environment for deploying the agent, background server tasks, web server, etc.

With such an environment we could automate builds and unit tests for integration with TravisCI, and other tools.

**Comment by atheurer** [comment]
I agree.  Related to that: since using pbench currently relies on a binary
packages to provide functionality (not just providing the agent, but
providing tools and benchmarks), we need to decide if this method is
mandatory, or if there should be an alternative.  'make install' is fine
-if it actually install everything (including all benchmarks and tools).
However, we will need to account for that method of use in our current code
("package" checking and installing).

**Comment by portante** [comment]
Most likely using `make install` will be preceded by a `./configure` step, and with that, other RPM packaging becomes a bit easier as well.  So this is worth doing for packaging reasons as well.

**Comment by ndokos** [comment]
On 07/15/2015 11:43 PM, Peter Portante wrote:
>
> Most likely using |make install| will be preceded by a |./configure| 
> step, and with that, other RPM packaging becomes a bit easier as well. 
> So this is worth doing for packaging reasons as well.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/distributed-system-analysis/pbench/issues/10#issuecomment-121818484>.
>

This is just for the pbench agent for now, right?

**Comment by portante** [comment]
Yes, and that begs the question, should we break up pbench into its subcomponents?

**Comment by ndokos** [comment]
On 07/16/2015 12:04 AM, Peter Portante wrote:
>
> Yes, and that begs the question, should we break up pbench into its 
> subcomponents?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/distributed-system-analysis/pbench/issues/10#issuecomment-121823146>.
>
I think we should.
